### PR TITLE
Dao  1

### PR DESCRIPTION
**Pull request by asthaaaa**
Added implementation of the methods storeLogLine and fileExists .
Added index template

### CODE REVIEW

**Comment by nishantnahata** [comment]
Please add PR description.

**Comment by asthaaaa** [comment]
Added implementation of the methods storeLogLine and fileExists .
Added index template

**Comment by shudhatmajain** [comment]
Split this PR into two : 
1. One for DAO.
2. Another for Document.java
They seemed like independent parts of code.

File: src/main/java/com/google/error_analyzer/backend/DaoInterface.java [hunk 1]
```diff
@@ -34,14 +34,9 @@
...
-    public String storeLogLine (String filename, String jsonString, String Id) 
+    //Stores the jsonString at index with name filename and returns the stored 
+    // string
+    public String storeLogLine (String filename, String jsonString, String id) 
     throws IOException;
 
-    //Stores the log into the database if an index with name fileName does not exist in the 
-    // database  and returns a string that contains the status of the log string whether the 
-    // log string was stored in the database or not.
-    public String checkAndStoreLog (String fileName, String log) throws IOException;
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/DaoInterface.java:45]: checkAndStoreLog
We don't need this function?

File: src/main/java/com/google/error_analyzer/backend/MockLogDao.java [hunk 2]
```diff
@@ -11,6 +11,8 @@
 
 package com.google.error_analyzer.backend;
 
+import com.google.error_analyzer.data.Document;
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/MockLogDao.java:14]: Sort imports alphabetically.

File: src/main/java/com/google/error_analyzer/backend/index_template.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/backend/index_template.json:1]: Where are we using this file?

File: src/main/java/com/google/error_analyzer/backend/DaoInterface.java (hunk 1 above)
Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/backend/DaoInterface.java:45]: This function will be included in LogDaoHelper.java

File: src/main/java/com/google/error_analyzer/backend/index_template.json (hunk 3 above)
Comment (asthaaaa) [reply src/main/java/com/google/error_analyzer/backend/index_template.json:1]: > Where are we using this file?

We will need to set this index template in elasticsearch

File: src/main/java/com/google/error_analyzer/data/Index.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+package com.google.error_analyzer.data;
+
+import com.google.error_analyzer.data.Document;
+import java.util.ArrayList;
+
+/*this class contains all the methods related to the indices of the mock 
+database. The Index class has attributes indexName and documentList.
+documentList is an Arraylist of objects of Document class.*/
+
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/data/Index.java:20]: Remove line space

File: src/main/java/com/google/error_analyzer/data/Document.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+limitations under the License.*/
+
+package com.google.error_analyzer.data;
+
+import java.util.ArrayList;
+
+/*The document class has attributes id and jsonString.
+ id represents the document id in the mock database.
+ Data will be stored as json strings in the documents*/
+
```
Comment (shudhatmajain) [inline src/main/java/com/google/error_analyzer/data/Document.java:19]: Remove line space

**Pull request merged**